framework/db/ActiveQueryTrait.php

```diff
@@ -159,7 +159,10 @@ private function createModels($rows)
      */
     public function findWith($with, &$models)
     {
-        $primaryModel = new $this->modelClass;
+        $primaryModel = reset($models);
+        if (!$primaryModel instanceof ActiveRecordInterface) {
+            $primaryModel = new $this->modelClass;
```

Comment:
we do not use it in any place inside AR currently so it would need to be changed in many places. I am thinking about moving the responsibility of creating relation objects to the AR class to fix #8639 but that is out of scope of this PR.